packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts

```diff
@@ -1347,7 +1347,8 @@ export class HttpRequestV3 implements INodeType {
 				const responseContentType = response.headers['content-type'] ?? '';
 				if (responseContentType.includes('application/json')) {
 					responseFormat = 'json';
-					response.body = jsonParse(Buffer.from(response.body).toString());
+					const data = Buffer.from(response.body).toString();
+					response.body = !data ? undefined : jsonParse(data);
```

Comment:
We should not be changing the default behavior here, as that would mean that if an API starts accidentally returning empty bodies, n8n would silently ignore the issue.

You could combine this check with the option `neverError`, to only ignore the error when a user has explicitly asked not to throw an error.

```suggestion
					const neverError = this.getNodeParameter(
						'options.response.response.neverError',
						0,
						false,
					) as boolean;
					const data = Buffer.from(response.body).toString();
					response.body = jsonParse(data, {
						...(neverError
							? { fallbackValue: {} }
							: { errorMessage: 'Invalid JSON in response body' }),
					});
```